tests/ut/py/test_worker/test_child_addr_guard.py

```
@@ -264,70 +269,130 @@ def test_alloc_child_records_then_free_clears(self):
+                w.copy_to(h, src, dst_offset=-1, nbytes=1)
+        assert nw.copy_to.call_count == 1
+
+    def test_copy_to_length_defaults_to_the_rest_of_the_host_side(self):
+        # The length has always come from the host side; an offset only moves where that side
+        # starts, so the no-nbytes call still means "the whole host backing".
+        w, nw = _l3_ready(0x2000)
+        h = w.alloc_child_tensor(0, (64,), DataType.UINT8)
+        with _host_buf(64) as src:
+            w.copy_to(h, src, src_offset=16)
+        assert nw.copy_to.call_args.args[3] == 48  # 64 - 16
+        with _host_buf(8) as src, pytest.raises(ValueError, match="src range .* exceeds the 8-byte backing"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)